cipher/rsa/rsa2.go

```diff
@@ -0,0 +1,125 @@
+/*
+rsa2.go
+description: RSA encryption and decryption including key generation
+details: [RSA wiki](https://en.wikipedia.org/wiki/RSA_(cryptosystem))
+author(s): [ddaniel27](https://github.com/ddaniel27)
+*/
+package rsa
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"math/rand"
+
+	"github.com/TheAlgorithms/Go/math/gcd"
+	"github.com/TheAlgorithms/Go/math/lcm"
+	"github.com/TheAlgorithms/Go/math/modular"
+	"github.com/TheAlgorithms/Go/math/prime"
+)
+
+// rsa struct contains the public key, private key and modulus
+type rsa struct {
+	publicKey  uint64
+	privateKey uint64
+	modulus    uint64
+}
+
+// New initializes the RSA algorithm
+// returns the RSA object
+func New() *rsa {
+	rsa := new(rsa)
+
+	// The following code generates keys for RSA encryption/decryption
+	// 1. Choose two large prime numbers, p and q and compute n = p * q
+	p, q := randomPrime() // p and q stands for prime numbers
+	modulus := p * q      // n stands for common number
+
+	// 2. Compute the totient of n, lcm(p-1, q-1)
+	totient := uint64(lcm.Lcm(int64(p-1), int64(q-1)))
+
+	// 3. Choose an integer e such that 1 < e < totient(n) and gcd(e, totient(n)) = 1
+	publicKey := uint64(2) // e stands for encryption key (public key)
+	for publicKey < totient {
+		if gcd.Recursive(int64(publicKey), int64(totient)) == 1 {
+			break
+		}
+		publicKey++
+	}
+
+	// 4. Compute d such that d * e ≡ 1 (mod totient(n))
+	inv, _ := modular.Inverse(int64(publicKey), int64(totient))
+	privateKey := uint64(inv)
+
+	rsa.publicKey = publicKey
+	rsa.privateKey = privateKey
+	rsa.modulus = modulus
+
+	return rsa
```

Comment:
Instead of creating a variable setting its values and returning, you can create an `rsa` value directly in the return statement.
```go
return rsa {
    publicKey: publicKey,
    privateKey: privateKey,
    modulus: modulus,
}
```

Reply:
Makes sense, done.